src/visit.rs

```diff
@@ -595,6 +595,53 @@ impl<'ast> Visit<'ast> for DiscoveryVisitor<'_> {
         };
         syn::visit::visit_expr_unary(self, i);
     }
+
+    fn visit_expr_match(&mut self, i: &'ast syn::ExprMatch) {
+        let _span = trace_span!("match", line = i.span().start().line).entered();
+
+        // While it's not currently possible to annotate expressions with custom attributes, this
+        // limitation could be lifted in the future.
+        if attrs_excluded(&i.attrs) {
+            trace!("match excluded by attrs");
+            return;
+        }
+
+        let has_catchall = i
+            .arms
+            .iter()
+            .any(|arm| matches!(arm.pat, syn::Pat::Wild(_)));
+        if has_catchall {
+            i.arms
+                .iter()
+                // Don't mutate the wild arm, because that will likely be unviable, and also
+                // skip it if a guard is present, because the replacement of the guard with 'false'
+                // below is logically equivalent to removing the arm.
```

Comment:
Actually this very good comment makes me wonder: if we have a match that has a default and that also one of the arms has a guard, do we redundantly generate mutants that delete the arm and also change the guard to `false`?

It looks like this code might do that today, but probably it should not, and also we should have a unit test for this case asserting that we don't.

Reply:
I think the case you're concerned about is exactly what I was explaining in the comment. I may not have added any tests for it, though.

Reply:
Yeah looks like I didn't.

Reply:
I added a test for it.